src/Common/src/Interop/Windows/kernel32/Interop.CreateFile2.cs

```diff
@@ -11,11 +11,58 @@ internal partial class Interop
     internal partial class Kernel32
     {
         [DllImport(Libraries.Kernel32, EntryPoint = "CreateFile2", SetLastError = true, CharSet = CharSet.Unicode, BestFitMapping = false)]
-        internal static extern SafeFileHandle CreateFile2(
+        internal unsafe static extern SafeFileHandle CreateFile2(
             string lpFileName,
             int dwDesiredAccess,
             System.IO.FileShare dwShareMode,
             System.IO.FileMode dwCreationDisposition,
             [In] ref CREATEFILE2_EXTENDED_PARAMETERS parameters);
+
+        private static unsafe SafeFileHandle CreateFile(
+            string lpFileName,
+            int dwDesiredAccess,
+            System.IO.FileShare dwShareMode,
+            Interop.Kernel32.SECURITY_ATTRIBUTES* securityAttrs,
+            System.IO.FileMode dwCreationDisposition,
+            int dwFlagsAndAttributes,
+            IntPtr hTemplateFile)
+        {
+            Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS parameters;
+            parameters.dwSize = (uint)Marshal.SizeOf<Interop.Kernel32.CREATEFILE2_EXTENDED_PARAMETERS>();
+
+            parameters.dwFileAttributes = (uint)dwFlagsAndAttributes & 0x0000FFFF;
+            parameters.dwSecurityQosFlags = (uint)dwFlagsAndAttributes & 0x000F0000;
+            parameters.dwFileFlags = (uint)dwFlagsAndAttributes & 0xFFF00000;
+
+            parameters.hTemplateFile = hTemplateFile;
+            parameters.lpSecurityAttributes = securityAttrs;
```

Comment:
Fixed not needed as it's copied into this stack based struct?

Reply:
Need to as it was passed in:

``` C#
        public unsafe static void Do(ref Guid guid)
        {
            // Complains
            DoNext(&guid);

            // This is ok
            Guid localGuid = new Guid();
            DoNext(&localGuid);
        }

        public unsafe static void DoNext(Guid* guid)
        {

        }
```